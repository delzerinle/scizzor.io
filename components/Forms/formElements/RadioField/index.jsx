import PropTypes from 'prop-types';
import { Field } from 'formik';

const RadioField = ({ name, label, className }) => {
  return (
    <div className={className}>
      <Field name={name}>
        {({ field, form }) => (
          <input
            id={name}
            {...field}
            type="checkbox"
            checked={field.value}
            onChange={() => {
              form.setFieldValue(field.name, !field.value);
              form.setFieldTouched(field.name, true);
            }}
            className="form-checkbox text-primary h-5 w-5 mr-3"
          />
        )}
      </Field>
      <label className="font-medium text-sm" htmlFor={name}>
        {label}
      </label>
    </div>
  );
};

RadioField.propTypes = {
  className: PropTypes.string,
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
};

export default RadioField;
