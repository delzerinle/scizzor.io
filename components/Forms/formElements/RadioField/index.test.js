import React from 'react';
import { shallow } from 'enzyme';

import RadioField from './index';

describe('RadioField tests', () => {
  it('renders without crashing', () => {
    shallow(<RadioField name="test" label="test" className="test" />);
  });

  it('radio snapshot', () => {
    const wrapper = shallow(
      <RadioField name="test" label="test" className="test" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
