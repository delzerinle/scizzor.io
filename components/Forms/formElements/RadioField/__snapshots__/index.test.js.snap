// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioField tests radio snapshot 1`] = `
<div
  className="test"
>
  <Field
    name="test"
  >
    <Component />
  </Field>
  <label
    className="font-medium text-sm"
    htmlFor="test"
  >
    test
  </label>
</div>
`;
